common/src/main/java/io/crate/types/TimestampType.java

```diff
@@ -108,4 +130,38 @@ public void writeValueTo(StreamOutput out, Long v) throws IOException {
     public int fixedSize() {
         return 16; // 8 object overhead, 8 long
     }
+
+    static long parseTimestamp(String timestamp) {
+        try {
+            return Long.valueOf(timestamp);
+        } catch (NumberFormatException e) {
+            return TimestampFormat.parseTimestampString(timestamp);
+        }
+    }
+
+    static long parseTimestampIgnoreTimeZone(String timestamp) {
```

Comment:
```suggestion
    static long parseTimestampWithOptionalTimeZone(String timestamp) {
```

Reply:
it is not optional, it ignores it. It parse it with or without time zone, but then converts it to UTC ignoring the time zone information.

Reply:
ah, no.. I read it wrong. It's fine as it is.